content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,846 @@
+----
+
+Note that ``` is available in EDN.
+
+* <<xref/../../reference/special_forms#quote,Clojure Official Documentation>>
+
+== `;` - Comment
+
+`;` is a comment. It takes all input from its starting point to the end of the
+line and ensures that the reader ignores it.
+[source,clojure]
+----
+user=> (def x "x") ; this is a comment
+#'user/x
+user=> ; this is a comment too
+<returns nothing>
+----
+
+== `:` - Keyword
+
+`:` is the indicator for a keyword which is an interned string that provides
+fast comparison and lower memory overhead.
+
+[source,clojure]
+----
+user=> (type :test)
+clojure.lang.Keyword
+----
+Alternatively you can use `keyword` to create a keyword from a string
+[source,clojure]
+----
+user=> (keyword "test")
+:test
+----
+A neat thing about keywords is that they alsoe implement `IFn` and can act as
+functions for extracting values from maps which is very nice:
+[source,clojure]
+----
+user=> (def my-map {:one 1 :two 2})
+#'user/my-map
+user=> (:one my-map) ; get the value for :one by invoking it as function
+1
+user=> (:three my-map) ; it can safely access non-keys
+nil
+user=> (:three my-map 3) ; it can return a default if specified
+3
+----
+
+* <<xref/../../reference/data_structures#Keywords,Clojure Official Documentation>>
+
+[[autoresolved_keys]]
+== `::` - Autoresolved keyword
+
+`::` is used to autoresolve a keyword to the current namespace:
+[source,clojure]
+----
+user=> :my-keyword
+:my-keyword
+user=> ::my-keyword
+:user/my-keyword
+user=> (= ::my-keyword :my-keyword)
+false
+----
+This is useful when creating macros. If you want to ensure that a macro that calls
+another function in the macro namespace correctly expands to call the function,
+you could use `::my-function` to refer to the fully qualified name.
+
+Note that `::` is not available in EDN
+
+* <<xref/../../reference/reader#,Reader>>
+
+== `/` - Namespace separator
+
+`/` can be the division function `clojure.core//`, but can also act as a
+separator in a symbol name to break apart the symbol name and the namespace it
+resides in, e.g. `my-namespace/utils`. This allows symbols to be fully qualified
+to prevent collisions.
+
+* <<xref/../../reference/reader#,Reader>>
+
+== `$` - Inner class reference
+
+Used to reference inner classes and interfaces in Java. Separates the
+container class name and the inner class name.
+[source,clojure]
+----
+(:import (basex.core BaseXClient$EventNotifier)
+
+(defn- build-notifier [notifier-action]
+  (reify BaseXClient$EventNotifier
+    (notify [this value]
+      (notifier-action value))))
+----
+
+`EventNotifier` is an inner interface of the `BaseXClient` class which is an
+imported Java class
+
+* http://blog.jayfields.com/2011/01/clojure-using-java-inner-classes.html[Clojure: Using Java Inner Classes]
+* <<xref/../../reference/java_interop#,Official Documentation>>
+
+== `->`, `->>`, `some->`, `cond->`, `as->` etc. - Threading macros
+
+These are threading macros. Almost all of them take an initial value and
```

Comment:
should refer to the Threading Macro guide https://clojure.org/guides/threading_macros